exchanges/order/order_test.go

```diff
@@ -2192,3 +2193,31 @@ func TestTrackingModeString(t *testing.T) {
 		require.Equal(t, v, k.String())
 	}
 }
+
+func TestUnmarshalOrder(t *testing.T) {
+	t.Parallel()
+	btx := currency.NewBTCUSDT()
+	btx.Delimiter = "-"
+	orderSubmit := Submit{
+		Exchange:  "test",
+		Pair:      btx,
+		AssetType: asset.Spot,
+		Side:      Buy,
+		Type:      Market,
+		Amount:    1,
+		Price:     1000,
+	}
+	jOrderSubmit, err := json.Marshal(orderSubmit)
+	require.NoError(t, err)
+	var os2 Submit
+	err = json.Unmarshal(jOrderSubmit, &os2)
+	require.NoError(t, err)
+	require.Equal(t, orderSubmit, os2)
+	// Margin-type regression test
+	orderSubmit.MarginType = margin.Multi
+	jOrderSubmit, err = json.Marshal(orderSubmit)
+	require.NoError(t, err)
+	err = json.Unmarshal(jOrderSubmit, &os2)
+	require.NoError(t, err)
+	require.Equal(t, orderSubmit, os2)
```

Comment:
Nits:
* Think the regression can be moved into the reason we do the test
* But actually, I don't think we should have this regression test here (in isolation of any other field tests); We didn't fix anything in order.Submit at all, so the regression wouldn't be here.
* Also: Am I missing why we couldn't have just set MarginType in the original orderSumbit and let L2215 handle this?
```suggestion
	orderSubmit.MarginType = margin.Multi
	jOrderSubmit, err = json.Marshal(orderSubmit)
	require.NoError(t, err, "json.Marshal must not error")
	err = json.Unmarshal(jOrderSubmit, &os2)
	require.NoError(t, err)
	require.Equal(t, orderSubmit, os2, "order.Submit must unmarshal Margin.Type correctly")
```

Reply:
I don't really know what you want me to do with this comment sorry. Do you want me to remove the whole test because its resolved in the `margin` package?

Reply:
Yeah. I guess I'm arguing for this 👇